rafthttp/peer.go

```diff
@@ -15,158 +15,169 @@
 package rafthttp
 
 import (
-	"errors"
+	"log"
 	"net/http"
-	"sync"
 	"time"
 
+	"github.com/coreos/etcd/Godeps/_workspace/src/golang.org/x/net/context"
 	"github.com/coreos/etcd/etcdserver/stats"
 	"github.com/coreos/etcd/pkg/types"
 	"github.com/coreos/etcd/raft/raftpb"
 )
 
 const (
-	appRespBatchMs = 50
-	propBatchMs    = 10
-
+	DialTimeout      = time.Second
 	ConnReadTimeout  = 5 * time.Second
 	ConnWriteTimeout = 5 * time.Second
+
+	recvBufSize = 4096
 )
 
 type peer struct {
```

Comment:
a peer is a remote raft node. local raft node sends message over peer.
a peer has two underlying mechanism to send out a message: stream and pipeline.
A stream is ... Peer also has a optimized stream for sending msgApp since...
A pipeline is ...